### PR TITLE
Support sending whatsapp template messages (Cloud API only)

### PR DESCRIPTION
**Pull request by SmittieC**
### Product Description
Adds support for sending WhatsApp template messages via the Meta Cloud API when the 24-hour service window has expired. When a bot tries to send a message to a user outside the service window, the system automatically falls back to a pre-configured template message, ensuring message delivery is not silently lost.

### Technical Description
- **New exception:** `ServiceWindowExpiredException` in `apps/chat/exceptions.py` signals when the WhatsApp 24h service window has expired.
- **Template message support in `MessagingService`:** The base class gains a `last_activity_at` parameter on `send_text_message` and `send_voice_message`. `MetaCloudAPIService` overrides `send_text_message` to detect expired windows and fall back to sending template messages via the Meta Cloud API.
- **Automatic template detection:** Removed the manual `has_template_message_configured` checkbox. The service now attempts to send a template and handles API errors gracefully if no template is configured.
- **Smart text splitting:** Uses `smart_split` from `telebot.util` to split long messages at word boundaries when they exceed the template parameter character limit (974 chars).
- **Voice fallback:** When voice delivery fails due to an expired service window, the system falls back to text, which in turn may fall back to a template message.
- **Form fields:** Added `template_language_code` (defaults to `"en"`) to the Meta Cloud API provider config form, with a `clean_template_language_code` method that defaults empty values to `"en"`.
- **Comprehensive tests:** 35 tests covering service window boundary conditions, template formatting, message splitting, error handling, voice-to-text fallback, and decorator propagation.

### Migrations
No migrations. All new state is stored in the existing service provider config dict.

### Docs and Changelog
- [x] This PR requires docs/changelog update

Docs should be updated to document the WhatsApp out-of-service-window template message flow:
- The system uses a template named `new_bot_message` with a body parameter named `bot_message` to send messages when the 24h service window has expired.
- Users who want their bot to support out-of-customer-window flows need to create a WhatsApp message template with that exact name and parameter in their Meta Business account, and get it verified/approved by Meta before it can be used.
- The template language code can be configured on the Meta Cloud API provider form (defaults to English).
- The template cannot use more than 100 characters (excluding the bot message)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/service_providers/tests/test_messaging_providers.py [hunk 1]
```diff
@@ -354,6 +358,288 @@ def test_send_voice_message(self, mock_post, meta_cloud_api_service):
...
+
+    def test_25_hours_ago_is_outside_window(self):
+        service = self._make_service()
+        last_activity = timezone.now() - timedelta(hours=25)
+        assert service._is_within_service_window(last_activity) is False
+
+    def test_exactly_24_hours_ago_is_outside_window(self):
+        service = self._make_service()
+        last_activity = timezone.now() - timedelta(hours=24)
+        assert service._is_within_service_window(last_activity) is False
```
[the remark this answers is not in the document]
Comment (SmittieC) [reply apps/service_providers/tests/test_messaging_providers.py]: Fixed in 5b900e0f4. Froze `timezone.now()` using `unittest.mock.patch` so the boundary test is deterministic.
